modules/widgets/top-posts.php

```diff
@@ -310,12 +314,31 @@ function widget( $args, $instance ) {
 		switch ( $display ) {
 		case 'list' :
 		case 'grid' :
+			// Keep the avatar_size as default dimensions for backward compatibility.
+			$width  = (int) $get_image_options['avatar_size'];
+			$height = (int) $get_image_options['avatar_size'];
+
+			// Check if the user has changed the width.
+			if ( ! empty( $get_image_options['width'] ) ) {
+				$width  = (int) $get_image_options['width'];
+			}
+
+			// Check if the user has changed the height.
+			if ( ! empty( $get_image_options['height'] ) ) {
+				$height = (int) $get_image_options['height'];
+			}
+
 			foreach ( $posts as &$post ) {
-				$image = Jetpack_PostImages::get_image( $post['post_id'], array( 'fallback_to_avatars' => true, 'avatar_size' => (int) $get_image_options['avatar_size'] ) );
+				$image = Jetpack_PostImages::get_image(
+					$post['post_id'],
+					array(
+						'fallback_to_avatars' => true,
+						'avatar_size'         => (int) $get_image_options['avatar_size']
```

Comment:
Since you're reorganizing this code, could you add a comma, since each line in an array declaration must end with a comma?